spock-core/src/main/groovy/spock/util/SourceToAstNodeAndSourceTranspiler.groovy

```diff
@@ -63,8 +68,15 @@ class SourceToAstNodeAndSourceTranspiler {
     CompilationUnit cu = new CompilationUnit((CompilerConfiguration)(config ?: CompilerConfiguration.DEFAULT), (CodeSource)codeSource.codeSource, (GroovyClassLoader)classLoader)
     def captureVisitor = new AstNodeCaptureVisitor()
     cu.addPhaseOperation(captureVisitor, compilePhase)
-    cu.addPhaseOperation(new AstNodeToScriptVisitor(writer, showSet), compilePhase)
     cu.addSource(codeSource.name, script)
+
+    boolean shallTraceBytecode = compilePhase >= CompilePhase.CLASS_GENERATION.phaseNumber
+    if (shallTraceBytecode) {
+      cu.setClassgenCallback(createClassgenCallback(writer, showSet))
+    } else {
+      cu.addPhaseOperation(new AstNodeToScriptVisitor(writer, showSet), compilePhase)
+    }
```

Comment:
This will have some impact on https://gwc-experiment.appspot.com/ as that is the default phase that is pre-selected, .